devito/passes/clusters/factorization.py

```diff
@@ -142,13 +142,16 @@ def run(expr):
                     terms.append(i)
 
             # Collect common funcs
-            w_funcs = Add(*w_funcs, evaluate=False)
-            w_funcs = collect(w_funcs, funcs, evaluate=False)
-            try:
-                terms.extend([Mul(k, collect_const(v), evaluate=False)
-                              for k, v in w_funcs.items()])
-            except AttributeError:
-                assert w_funcs == 0
+            if len(w_funcs) > 1:
```

Comment:
@FabioLuporini  this does not fix the pickle reconstruction issue with `(-1)*` so left the `_print_Mul` in printer for the time being

Reply:
OK. Can you open an issue about that and refer to the failing test without the hack?